### DISCUSSION TITLE
Query to select a row of table returns null and undefined

### DISCUSSION THREAD

**Discussion by miladsade96** [supabase/supabase#17265] [Questions]
I have a react app which uses supabase as its database.
this is my table:
![Settings_Table](https://github.com/supabase/supabase/assets/57490926/ad2bf0b3-dbb0-489d-8f83-0785da35dabb)
I have set the RLS policies to read and update for all users.
In the other hand, this is my react app codes:
Query function:
![getSettings](https://github.com/supabase/supabase/assets/57490926/1ce5a589-fe03-4e90-b7f2-4287bb6bdaec)
Custom hook:
![useSettings](https://github.com/supabase/supabase/assets/57490926/874ca5a4-419e-49c4-9e6b-506f5c415574)
and my functional component:
![UpdateSettingsForm](https://github.com/supabase/supabase/assets/57490926/cacec4d7-bb44-46c9-b16e-a4901b33c452)
But i get this error:
![error](https://github.com/supabase/supabase/assets/57490926/19c04748-5b4b-4f63-b76b-89a4abe7e05e)

**Comment by MildTomato** [comment]
@JS-Mechanic The errors seem to be in relation to the Form component you have, rather than the supabase query you have setup. 

Have you console logged `data` and `error` in your getSettings function?
might be good to check if anything is being returned from that first

The `.single()` method might also be getting misused because if you have multiple rows in `settings` table you will have an array of objects returned.

It will be easier for others to help if we can look at a full code example, got a GitHub link, codesandbox link or something similar?

**Reply by miladsade96** [reply]
Hi @MildTomato 
I just realized that i got the values from supabase, But i encounter with an error whenever i use <Form> component of react-hook-form library. I have no idea what is wrong with that because i used it before in my app without any problem.

**Reply by MildTomato** [reply]
> I just realized that i got the values from supabase, But i encounter with an error whenever i use

I'm not really following what you mean. It returned the values but you had an error?

**Reply by MildTomato** [reply]
> I have no idea what is wrong with that because i used it before in my app without any problem.

It looks like you are also using TanStack React Query.

I think what would be best is you could share a working version of the files so the community here can help debug whatever the issue is. It also looks like maybe it's a better question for https://github.com/TanStack/query/discussions if your having issues using the React Query library.
